docs/fastdds/transport/zero_copy/zero_copy.rst

```diff
@@ -0,0 +1,161 @@
+.. include:: ../../../03-exports/aliases.include
+.. include:: ../../../03-exports/aliases-api.include
+.. include:: ../../../03-exports/roles.include
+
+.. _transport_zero_copy:
+
+Zero-Copy
+=========
+
+This section explains the Zero-Copy transfer mode implemented in *Fast DDS*.
+The Zero-Copy transfer allows the transmission of data between DDS |DataWriter| and |DataReader| entities without
+making copies of the data being transmitted.
+These samples are stored in a shared memory region, which is accessible by multiple processes implementing an
+application using Fast DDS. In other words, the DataWriter has its History in a memory mapped file accessible to
+the DataReader.
+
+The figure below shows a comparison between the different transports available in Fast DDS.
+
+.. figure:: /01-figures/fast_dds/transport/transport_comparison.png
+    :align: center
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+---------
+
+When creating a DataWriter that supports Zero-Copy transfers, samples must be created with a *Fast DDS* function
+that extends the DDS API (:func:`loan_sample`).
+The return of this function is a reference A* to the sample being sent, that is, a reference to the sample stored in the
+memory mapped file.
+The reference to this sample is sent to the DataReader which supports Zero-Copy and which is attached to the memory
+mapped file.
+Thus, the user has access to a reference B* to the sample.
+Since both processes do not share process memory, i.e., each has an assigned virtual process memory,
+the references A* and B* will be different but both point to the same RAM region.
+
+It is worth mentioning that one of the main advantages of this transfer solution is to avoid encapsulation of
+RTPS packets for DDS entities running on the same machine.
+
+This feature requires the usage of new *Fast DDS* API which extends the standard DDS API.
+
+* :func:`DataWriter::loan_sample`
+* :func:`DataWriter::discard_loan`
+
+Getting started
+---------------
+
+To enable Zero-Copy perform the following steps:
+
+1.  Define a plain and bounded type in an IDL file and generate the corresponding source code for further processing
+    with the |Fast DDS-Gen| tool.
+
+    .. code-block:: idl
+
+        struct LoanableHelloWorld
+        {
+            unsigned long index;
+            char message[256];
+        };
+
+2.  On the DataWriter side:
+
+    a)  Create a DataWriter for the previous type.
+    b)  Get a loan on a sample using :func:`loan_sample`.
+    c)  Write the sample using |DataWriter::write-api|.
```

Comment:
`Make sure that the DataWriter does not  have DataSharing disabled.`
(with a reference to the DataSharing documentation when merged)

Reply:
:+1: